extensions/markdown-language-features/src/languageFeatures/copyFiles/shared.ts

```diff
@@ -56,9 +56,36 @@ export const mediaMimes = new Set([
 	'audio/x-wav',
 ]);
 
+export async function getMarkdownLink(document: vscode.TextDocument, _ranges: readonly vscode.Range[], dataTransfer: vscode.DataTransfer, uriEdit: vscode.DocumentPasteEdit, token: vscode.CancellationToken) {
+	if (_ranges.length === 0) {
+		return;
+	}
+
+	const selectedTexts = _ranges.map(range => document.getText(range));
+	const rangeTextTuples = _ranges.map((range, index) => [range, selectedTexts[index]] as [vscode.Range, string]);
+	const edits: vscode.SnippetTextEdit[] = [];
+	let count: number;
+	count = rangeTextTuples.length;
+
+	for (const [range, text] of rangeTextTuples) {
+		const snippet = await tryGetUriListSnippet(document, dataTransfer, token, text, count);
+		if (!snippet) {
+			return;
+		}
+		count--;
+		edits.push(new vscode.SnippetTextEdit(range, snippet.snippet));
+		uriEdit.label = snippet.label;
+	}
+
+	const newAdditionalEdits = new vscode.WorkspaceEdit();
+	newAdditionalEdits.set(document.uri, edits);
+	uriEdit.additionalEdit = newAdditionalEdits;
+	return uriEdit;
```

Comment:
Instead of modifying uriEdit, can you return info that the caller can user to create a uri edit? This is more like what `tryGetUriListSnippet` does